1.regression_modelling/scripts/1.train_models.py

```diff
@@ -0,0 +1,213 @@
+elasticnet_params = {
+    "alpha": 1.0,  # Equivalent to 'C' in LogisticRegression, but in reverse
+    "l1_ratio": 0.5,  # Mixture of L1 and L2 regularization
+    "max_iter": 10,
+    "random_state": 0,
+}
+
+# Define the hyperparameter search space for RandomizedSearchCV
+param_dist = {
+    "alpha": np.logspace(-3, 3, 7),  # Regularization strength
+    "l1_ratio": np.linspace(0, 1, 11),  # Mix of L1 and L2 regularization
+}
+
+# Set the random search hyperparameterization method parameters
+random_search_params = {
+    "param_distributions": param_dist,
+    "scoring": "neg_mean_squared_error",  # Suitable for regression
+    "random_state": 0,
+    "n_jobs": -1,
+    "cv": k_folds,
+}
+
+
+# ## Train A647 models
+
+# In[5]:
+
+
+# Suppress all warnings
+warnings.filterwarnings("ignore")
+
+for a647_feature in a647_features:
+    y = training_df[a647_feature]
+
+    # Train regular model with hyperparameter tuning
+    logreg = ElasticNet(**elasticnet_params)
+
+    # Initialize random search and fit model
+    random_search = RandomizedSearchCV(logreg, **random_search_params)
+    random_search.fit(X, y)
+
+    # Save the tuned model
+    model_filename = final_dir / f"{a647_feature}_tuned_model.joblib"
+    joblib.dump(random_search.best_estimator_, model_filename)
+
+    random_search_shuffled = RandomizedSearchCV(logreg, **random_search_params)
+    random_search_shuffled.fit(X_shuffled, y)
+
+    # Save the shuffled tuned model
+    shuffled_model_filename = (
+        shuffled_dir / f"{a647_feature}_shuffled_tuned_model.joblib"
+    )
+    joblib.dump(random_search_shuffled.best_estimator_, shuffled_model_filename)
+
+    # Print confirmation
+    print(f"Trained and saved tuned model for {a647_feature}")
+    print(f"Trained and saved shuffled tuned model for {a647_feature}")
+
+print("All A647 models have been trained and tuned!")
+
+
+# # Train GOLD models
+
+# In[6]:
+
+
+for gold_feature in gold_features:
+    y = training_df[gold_feature]
+
+    # Train regular model with hyperparameter tuning
+    logreg = ElasticNet(**elasticnet_params)
+
+    # Initialize random search
+    random_search = RandomizedSearchCV(logreg, **random_search_params)
+    random_search.fit(X, y)
+
+    # Save the tuned model
+    model_filename = final_dir / f"{gold_feature}_tuned_model.joblib"
+    joblib.dump(random_search.best_estimator_, model_filename)
+
+    random_search_shuffled = RandomizedSearchCV(logreg, **random_search_params)
+    random_search_shuffled.fit(X_shuffled, y)
+
+    # Save the shuffled tuned model
+    shuffled_model_filename = (
+        shuffled_dir / f"{gold_feature}_shuffled_tuned_model.joblib"
+    )
+    joblib.dump(random_search_shuffled.best_estimator_, shuffled_model_filename)
+
+    # Print confirmation
+    print(f"Trained and saved tuned model for {gold_feature}")
+    print(f"Trained and saved shuffled tuned model for {gold_feature}")
+
+print("All GOLD models have been trained and tuned!")
```

Comment:
Consider adding the gold and A647 feature lists together in one list since they both use the same logreg params.
This would reduce code and DNR,

Reply:
This is more my personal preference and wanting to make sure that I have each set of features separate, I will keep with even though DNR is happening for now. I think for another PR this can be update, thanks for bringing it up!